src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -28,6 +28,32 @@ constexpr auto processorStateKey  = wxT("ProcessorState");
 constexpr auto controllerStateKey = wxT("ControllerState");
 constexpr auto parametersKey = wxT("Parameters");
 
+struct VST3PluginCache
+{
+   Steinberg::Vst::ParamID rootUnitProgramChangeParameterID { Steinberg::Vst::kNoParamId };
+   Steinberg::Vst::ProgramListID rootUnitProgramListID { Steinberg::Vst::kNoProgramListId };
+   Steinberg::int32 rootUnitProgramCount { 0 };
+};
+
+std::map<std::string, VST3PluginCache> sVST3PluginCache;
+
+VST3PluginCache* GetCache(const VST3::UID& effectUid)
+{
+   const auto key = effectUid.toString();
+   auto it = sVST3PluginCache.find(key);
+   if(it != sVST3PluginCache.end())
+      return &it->second;
+   return nullptr;
+}
+
+VST3PluginCache& CreateCache(const VST3::UID& effectUid)
+{
+   const auto key = effectUid.toString();
+   assert(sVST3PluginCache.find(key) == sVST3PluginCache.end());
+   const auto result = sVST3PluginCache.insert(std::pair { key, VST3PluginCache {} });
+   return result.first->second;
```

Comment:
Why not simply combine the two functions ?  Just let `GetCache` return a reference to `sVST3PluginCache[key]` and that will create and default-initialize a member that isn't there already.

Then the assertion isn't needed, and you know I don't like assertions without comments:  when I review an assertion,. I'm not satisfied until I also review all uses of the function.  Less burden of review without it.

Reply:
The answer is in VST3Wrapper constructor - cache initialization process. If `GetCache` and `CreateCahce` were combined I had to add an initialization state. Reference type might have convinced you that return value isn't nullptr (though that can be violated too), and the code is formally correct, but you must make sure that the state of returned object is correct too. I don't think that it worth changing.